app/bundles/LeadBundle/Model/FieldModel.php

```diff
@@ -374,46 +374,54 @@ public function saveEntity($entity, $unlock = true)
             $entity->setIsListable(false);
         }
 
-        $event = $this->dispatchEvent('pre_save', $entity, $isNew);
-        $this->getRepository()->saveEntity($entity);
-        $this->dispatchEvent('post_save', $entity, $isNew, $event);
-
         $isUnique = $entity->getIsUniqueIdentifier();
 
-        if ($entity->getId()) {
-            //create the field as its own column in the leads table
-            $leadsSchema = $this->schemaHelperFactory->getSchemaHelper('column', $object);
-            if ($isNew || (!$isNew && !$leadsSchema->checkColumnExists($alias))) {
-                $schemaDefinition = self::getSchemaDefinition($alias, $entity->getType(), $isUnique);
-                $leadsSchema->addColumn(
+        //create the field as its own column in the leads table
+        $leadsSchema = $this->schemaHelperFactory->getSchemaHelper('column', $object);
+        if ($isNew || (!$isNew && !$leadsSchema->checkColumnExists($alias))) {
+            $schemaDefinition = self::getSchemaDefinition($alias, $type, $isUnique);
+            $leadsSchema->addColumn(
                     $schemaDefinition
                 );
+
+            try {
                 $leadsSchema->executeChanges();
+                $isCreated = true;
+            } catch (\Exception $e) {
+                $this->logger->addWarning($e->getMessage());
+                $isCreated = false;
+                throw new \ErrorException('Too many field');
```

Comment:
I agree we should probably change the exception to something more descriptive though :joy: